service/matching/db_task_manager.go

```diff
@@ -0,0 +1,258 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package matching
+
+import (
+	"sync/atomic"
+	"time"
+
+	enumspb "go.temporal.io/api/enums/v1"
+
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/future"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/service/worker/scanner/taskqueue"
+)
+
+const (
+	dbTaskFlushInterval = 24 * time.Millisecond
+
+	dbTaskUpdateAckInterval   = time.Minute
+	dbTaskUpdateQueueInterval = time.Minute
+)
+
+type (
+	dbTaskManager struct {
```

Comment:
should be taskqueueManager or taskManager?

Reply:
no, this is managing the task which are going to be write to DB and read back

Reply:
But this is one instance per task queue. I'm trying to reduce new concept that people need to understand. Reuse the existing name would help folks who already understand current codebase.

Reply:
>But this is one instance per task queue

this is per task queue, but managing write / read to DB.

>I'm trying to reduce new concept that people need to understand. 

this has nothing to do with implementation. implementation itself should be as clean as possible.